metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/util/IcebergFilterGenerator.java

```diff
@@ -200,4 +130,121 @@ private Expression evalSingleTerm(final ASTCOMPARE node, final Object data) {
         }
         return Expressions.alwaysFalse();
     }
+
+    /**
+     * evalString.
+     *
+     * @param node node
+     * @param data data
+     * @return eval String
+     */
+    private Expression evalString(final SimpleNode node, final Object data) {
+        final Object lhs = node.jjtGetChild(0).jjtAccept(this, data);
+        final Compare comparison = (Compare) node.jjtGetChild(1).jjtAccept(this, data);
+        final Object rhs = node.jjtGetChild(2).jjtAccept(this, data);
+        return createIcebergExpression(lhs, rhs, comparison);
+    }
+
+    /**
+     * Check if the key is part of field.
+     *
+     * @param key input string
+     * @return True if key is a field.
+     */
+    private boolean isField(final Object key) {
+        return (key instanceof String) && fieldMap.containsKey(((String) key).toLowerCase());
+    }
+
+    /**
+     * Check if the key is an iceberg supported date filter field.
+     *
+     * @param key input string
+     * @return True if key is an iceberg supported date filter field.
+     */
+    private boolean isIcebergTimestamp(final Object key) {
+        return (key instanceof String) && ICEBERG_TIMESTAMP_NAMES.contains(key);
+    }
+
+    /**
+     * Get the key and value field of iceberg expression.
+     *
+     * @param lhs left hand string
+     * @param rhs right hand string
+     * @return key value pair for iceberg expression.
+     */
+    private Pair<String, Object> getExpressionKeyValue(final Object lhs,
+                                                       final Object rhs) {
+        if (isIcebergTimestamp(lhs)) {
+            return new ImmutablePair<>(lhs.toString(), ((BigDecimal) rhs).longValue());
```

Comment:
Any reason using ImmutablePair instead of Map.Entry?

Reply:
The pair is more appropriate here since it can only have two values left hand value and right hand value.